### PR TITLE
NEXT-10847 - Add basic management of scheduled tasks

### LINKED ISSUE #104: NEXT-10847 - Overview and basic management of scheduled tasks from the administration
**Description**  

Currently there is no way to see and manage scheduled tasks except for looking at the database itself. As a first step one can list all scheduled tasks and add the possiblity to modify next execution time, change the interval and status, similar to how it was possible in SW5.

### PR DESCRIPTION
**Pull request by felixbrucker**
### 1. What does this change do, exactly?
This PR adds a settings module to view, edit and manually run scheduled tasks:

ezgif com-video-to-gif(1)

### 2. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [x] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by jleifeld** [comment]
Changelog is missing :)

**Comment by felixbrucker** [comment]
Yup @jleifeld, just added some last changes as well as the changelog and markd the PR as ready for review

File: src/Core/Framework/MessageQueue/ScheduledTask/Exception/ScheduledTaskException.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\MessageQueue\ScheduledTask\Exception;
+
+class ScheduledTaskException extends \Exception
```
Comment (mitelg) [inline src/Core/Framework/MessageQueue/ScheduledTask/Exception/ScheduledTaskException.php:5]: extend from ShopwareHttpException

**Review by shyim — APPROVED** [review]
For PHP is anything okay for me

File: src/Core/Framework/MessageQueue/ScheduledTask/Exception/ScheduledTaskNotFoundException.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace Shopware\Core\Framework\MessageQueue\ScheduledTask\Exception;
+
+use Shopware\Core\Framework\ShopwareHttpException;
+
+class ScheduledTaskNotFoundException extends ShopwareHttpException
+{
+    public function __construct(string $scheduledTaskId)
+    {
+        parent::__construct(sprintf('No scheduled task found for id %s', $scheduledTaskId), [
```
Comment (mitelg) [inline src/Core/Framework/MessageQueue/ScheduledTask/Exception/ScheduledTaskNotFoundException.php:11]: we have some kind of templating within the ShopwareHttpException
```suggestion
        parent::__construct('No scheduled task found with ID "{{ scheduledTaskId }}"', [
```
just use the array key of the second parameters you give

File: src/Core/Framework/MessageQueue/ScheduledTask/ScheduledTaskService.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        $this->scheduledTaskRepository = $scheduledTaskRepository;
+        $this->taskHandler = $taskHandler;
+    }
+
+    public function runScheduledTask(string $scheduledTaskId, Context $context): void
+    {
+        $criteria = new Criteria([$scheduledTaskId]);
+        /** @var ScheduledTaskEntity $scheduledTask */
+        $scheduledTask = $this->scheduledTaskRepository->search($criteria, $context)->first();
+        if (!$scheduledTask) {
```
Comment (mitelg) [inline src/Core/Framework/MessageQueue/ScheduledTask/ScheduledTaskService.php:36]:
```suggestion
        if ($scheduledTask === null) {
```

**Review by mitelg — CHANGES_REQUESTED** [review]

**Review by mitelg — APPROVED** [review]

**Comment by mitelg** [comment]
someone with JS skills should also have a look here @shopwareBoostDay/shopware-devs

**Review by raknison — APPROVED** [review]
The JS part is okay so far. I have nothing to complain about except that a few tests would be nice :-)

**Pull request merged**